api/v1alpha1/clusterspiffeid_webhook.go

```diff
@@ -129,7 +130,17 @@ func ParseClusterSPIFFEIDSpec(spec *ClusterSPIFFEIDSpec) (*ParsedClusterSPIFFEID
 	}
 
 	var dnsNameTemplates []*template.Template
+	dnsNameTemplatesUnique := make(map[string]struct{})
 	for _, value := range spec.DNSNameTemplates {
+		// Check for duplicate templates. Strip out white space as the template can have variable white
+		// spaces between brackets, ie "{{ .PodMeta.Name }}" and "{{.PodMeta.Name}}" are the same
+		dnsNameTemplateNoSpaces := strings.ReplaceAll(value, " ", "")
+		if _, exists := dnsNameTemplatesUnique[dnsNameTemplateNoSpaces]; exists {
+			return nil, fmt.Errorf("duplicate dnsNameTemplate: %s", value)
```

Comment:
I'm not sure this validation is useful since it's possible to still get duplicates otherwise. Since we're going to dedup the names anyway, what's the value add here?

Reply:
I was thinking catching the error earlier, ie if the duplicate template is a typo.

Reply:
Removed this check